lib/job/arUpdateDescendantsJob.class.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ *
+ *
+ * @package    symfony
+ * @subpackage jobs
+ */
+
+class arUpdateDescendantsJob extends arBaseJob
```

Comment:
I think we could improve a lot this job to accept other big ES documents updates without a big amount of changes.

I'd do the following:
- Rename the job to arUpdateIoEsDocumentsJob (or similar).
- Require the following parameters:
  - An array of IO ids.
  - A boolean to update the IOs from the array or not.
  - A boolean to update their descendants or not.
- Change the process logic to:
  - Update the IOs from the array based on the first boolean.
  - Update their descendants based on the second boolean.

That way we could use this job in this case and in other big IO updates, like when a term or an actor is updated:

https://github.com/artefactual/atom/blob/qa/2.5.x/apps/qubit/modules/term/actions/editAction.class.php#L578-L589
https://github.com/artefactual/atom/blob/qa/2.5.x/lib/model/QubitActor.php#L146-L153

Reply:
Ideally, this job could accept a partial update option, where only some fields are updated instead of the entire IOs, but that could be done in the future. This option should only be considered in the IOs from the array, descendants should always use partial updates, like it's being done.

Later, we could rename and add other resources to this job, like QubitActor and QubitTerm.

Reply:
Feel free to disregard this if running out of time. We can create a new feature ticket to not forgot about the enhancement in the future.